### PR TITLE
Feat/288 update tool hierarchy

### LINKED ISSUE #281: Irrelevant Label NoData Value field in Predict Classifier and Predict Regressor tools
Label NoData Value field is not needed as labels are not utilized for prediction.

### LINKED ISSUE #282: Hinge loss on Train MLP Regressor
Primary use of hinge loss is in classification tasks, it should not be available for Train MLP Regressors loss function

### LINKED ISSUE #287: MLP training should not allow a dropout rate of 1
Currently `_check_MLP_inputs` validates that `0.0 <= dropout_rate <= 1.0`, when it should in fact check `0.0 <= dropout_rate < 1.0`.

### LINKED ISSUE #288: Update tool hierarchy
Update tool hierarchy according to the picture below:

Image

### PR DESCRIPTION
**Pull request by em-t**
Updates according to latest specs from Juho and Johanna:
- Update tool hierarchy. Closes #288 
- Remove some more tools (TOC Fuzzification and Fuzzy ROC).
- Rename Thinning and Splitting tools + some of their input fields.

Took the opportunity to do some minor fixes at the same time:
- Remove hinge loss function from Train MLP Regressor. Closes #282 
- Fix MLP training dropout rate accepted value range. Closes #287 
- Remove Label NoData Value field from MLP prediction tools. Closes #281 
- Minor clean-up

### CODE REVIEW

**Review by lehtonenp — APPROVED** [review]
LGTM in general and is G2G (good to go).

My note about hierarchy. `Preprocessing -> Raster Processing -> Fuzzy Membership` path leads to just `Categorical & Reclass`. Should it be either of the two:
- `Preprocessing -> BOTH RASTER AND VECTOR TOOLS`
- `Preprocessing -> Raster Processing -> Categorical & Reclass`

**Comment by em-t** [comment]
> LGTM in general and is G2G (good to go).
> 
> My note about hierarchy. `Preprocessing -> Raster Processing -> Fuzzy Membership` path leads to just `Categorical & Reclass`. Should it be either of the two:
> 
> * `Preprocessing -> BOTH RASTER AND VECTOR TOOLS`
> * `Preprocessing -> Raster Processing -> Categorical & Reclass`

Yeah, I was wondering about this as well. @jtlait do you have an opinion?

**Comment by jtlait** [comment]
@lehtonenp and @em-t you are correct, the naming is not good. I think that separate raster and vector submenus are intuitive and I am in favor to have them there to guide the user. Fuzzy membership could be removed but then perhaps better naming should be invented as categorical & reclass is connected to fuzzy membership values. Any suggestions how to rename categorical & reclass? I will also look more closely what the tool does and propose something.

**Comment by lehtonenp** [comment]
Ok. If @jtlait suggest a better naming for `Categorical & Reclass` and then we move it up in the hierarchy by one. How does that sound?

So, the hierarchy would be the following:
- `Preprocessing -> Raster Processing -> Newly Titled Categorical & Reclass`

**Comment by jtlait** [comment]
@lehtonenp, @em-t sounds good. I will suggest a new name of `Reclass & Fuzzify` for the `Categorical & Reclass`. In my opinion, it lines up with the tool functionality well.
